storybook/pagesmodel.cpp

```diff
@@ -0,0 +1,90 @@
+#include "pagesmodel.h"
+
+#include <QRegularExpression>
+#include <QDir>
+
+namespace {
+const auto categoryFirst QStringLiteral("_");
+const auto categoryUncategorized QStringLiteral("Uncategorized");
+}
+
+PagesModel::PagesModel(const QString &path, QObject *parent)
+    : QAbstractListModel{parent}
+{
+    QDir dir(path);
+    dir.setFilter(QDir::Files);
+
+    static QRegularExpression fileNameRegex(
+                QRegularExpression::anchoredPattern("(.*)Page\\.qml"));
+    static QRegularExpression categoryRegex(
+                "^//(\\s)*category:(.+)$", QRegularExpression::MultilineOption);
+
+    const QFileInfoList files = dir.entryInfoList();
+
+    std::for_each(files.begin(), files.end(), [this] (auto &fileInfo) {
+        QString fileName = fileInfo.fileName();
+        QRegularExpressionMatch fileNameMatch = fileNameRegex.match(fileName);
+
+        if (!fileNameMatch.hasMatch())
+            return;
+
+        QFile file(fileInfo.filePath());
+        file.open(QIODevice::ReadOnly);
+        QByteArray content = file.readAll();
```

Comment:
This is... meh :) Imagine we have a 1000 files here and you'll go about opening and reading all of their contents just to get the some metadata from them :/

Reply:
It's an internal tool with a clearly defined purpose. We know that we will handle only qml source files here. And it's clear that tools like that should follow KISS first instead of try to do premature optimizations.
And instead of imaging it's better to measure and rely on data. I tested in on reading 8K qml files with regex matching on them.:
- linux: ~70ms
- win: ~250ms

In current pace we will reach that amount in more than 20 years. Then we can consider more sophisticated solution and invest more time in that :)